awx/ui/test/e2e/tests/test-users-crud.js

```diff
@@ -0,0 +1,54 @@
+/* Tests for the user CRUD operations. */
+import { genID } from '../utils';
+
+const testID = genID();
+
+const store = {
+    organization: {
+        name: `org-${testID}`
+    },
+    user: {
+        username: `user-${testID}`,
+        email: `email-${testID}@example.com`,
+        password: `${testID}`,
+    },
+};
+
+module.exports = {
+    before: client => {
+        client.login();
+        client.waitForAngular();
+
+        client.inject(
+            [store, 'OrganizationModel'],
+            (_store_, Model) => new Model().http.post({ data: _store_.organization }),
+            ({ data }) => { store.organization = data; }
+        );
+    },
+    'create an user': client => {
+        const users = client.page.users();
+        users.load();
+        users.section.list
+            .waitForElementVisible('@add')
+            .click('@add');
+        users.section.add
+            .waitForElementVisible('@title')
+            .setValue('@organization', store.organization.name)
+            .setValue('@email', store.user.email)
+            .setValue('@username', store.user.username)
+            .setValue('@password', store.user.password)
+            .setValue('@confirmPassword', store.user.password)
+            .click('@save');
+        client.waitForSpinny();
+        users.section.list.section.search
+            .setValue('@input', store.user.username)
+            .click('@searchButton');
+        client.waitForSpinny();
+        const row = '#users_table .List-tableRow';
+        users.waitForElementNotPresent(`${row}:nth-of-type(2)`);
```

Comment:
The UI will, I believe, remove _all_ rows before populating new ones after a search. To avoid a race condition, you should probably also add a `waitForElementPresent(row)` before doing any assertions on its text content.

Reply:
I found this way to wait for the search to complete from other tests and I did not have any issue so far waiting for the proper elements to be on the page. Do you still want me to update this? Thanks

Reply:
Alright - if it's already working let's leave it as is.